zkvm/src/blockchain/state.rs

```diff
@@ -62,6 +63,17 @@ impl BlockchainState {
 
         Ok(())
     }
+
+    fn apply_txlogs<I>(&mut self, txlist: I) -> Result<(), VMError>
+    where
+        I: Iterator,
```

Comment:
IntoIterator is even better, since the user can send their Vec or &Vec directly w/o doing `.iter()`.
```suggestion
        I: IntoIterator,
```